### PR TITLE
Docs/3306/restructure-analysis-docker-docs

### PR DESCRIPTION
**Pull request by moritz-suckow**
# Expand and restructure Docker usage documentation

Issue: #3306 

## Description

This PR expands and restructures the documentation regarding the usage of Docker in CodeCharta. 
Examples on how to use CodeCharta without docker-compose are added.

### CODE REVIEW

File: analysis/README.md [hunk 1]
```diff
@@ -61,7 +70,20 @@ This installs all binaries to run the analysis. Java 11 is recommended, while Ja
...
-## Installation
+## Installation via Docker
+
+You can use this via docker compose or as a standalone container. This section will deal with how to use the analysis as a standalone container. For information on how to use this with docker compose, please check out [Docker Getting Started]({{site.baseurl}}{% link _docs/01-04-docker-containers.md %})
+We assume that you already installed docker, if not, you have to do that before!
+
+To containerize the analysis, please follow the below listed steps.
+
+-   Navigate into the directory you want to analyse with CodeCharta. There are multiple ways to use the docker image:
+    1. Start the docker container and a bash shell in it with `docker run -it -v $(pwd):$(pwd) -w $(pwd) codecharta/codecharta-analysis bash` (for that to work you have to make sure bash is added to your PATH). This runs the image and mounts the current directory and sets it as the working directory in the image. You can now use the CodeCharta shell or any other of the installed tools via the command line.
```
Comment (ce-bo) [inline analysis/README.md:81]: By the stated docker command, the bash command is executed with in a running instance of the image (called container). So there is no need that the users have bash installed on their local system (including setting path variables).

Furthermore, I would add a sentence like this: You can use all the commands availalable in the container <Link to Docker Image or List of commands>. Do we have a list of available commands documented somewhere?

File: visualization/README.md [hunk 2]
```diff
@@ -19,6 +19,8 @@ You can start with **Codecharta Visualization** on multiple ways:
 -   Installation as a [npm package](#npm-package) (Recommended)
 -   Run an operating system specific [standalone](#run-a-standalone)
 -   Build it yourself from the github [repository](#build-it-yourself)
+-   Use [docker-compose]({{site.baseurl}}{% link _docs/01-04-docker-containers.md %}) to run this with all other containers
```
Comment (ce-bo) [inline visualization/README.md:22]:
```suggestion
-   Use [docker-compose]({{site.baseurl}}{% link _docs/01-04-docker-containers.md %}) to run this with other needed tools like a Sonar instance or a analyzing tools of codecharta-analysis.
```

File: visualization/README.md [hunk 3]
```diff
@@ -140,6 +142,16 @@ $ npm run e2e:auto
...
 > To follow/watch the steps the e2e test is performing, deactivate headless mode in `jest-puppeteer.config.js` (and maybe set the `slowMo` parameter)
 
+## Run in Docker container
+
+You can use this via docker compose or as a standalone container. This section will deal with how to use the visualization as a standalone container. For information on how to use this with docker compose, please check out [Docker Getting Started]({{site.baseurl}}{% link _docs/01-04-docker-containers.md %})
+We assume that you already installed docker, if not, you have to do that before!
+
+To containerize the visualization, please follow the below listed steps.
+
+-   Start the docker container: `docker run -d -p 80:9000 codecharta/codecharta-visualization`. This detaches the container and exposes port 80 on the container and port 9000 on the host.
```
Comment (ce-bo) [inline visualization/README.md:152]: This command will provide the visualization localhost:80 on the users machine. I think the port mapping must be turned around.

File: analysis/README.md (hunk 1 above)
Comment (moritz-suckow) [reply analysis/README.md:81]: Thanks for the info!

Right now I did not find a list with all available commands. I can add a list with all tools and how you can find out more about them (like the help command for each major tool/command)

**Comment by moritz-suckow** [comment]
I also fixed the broken badge in the Visualization Readme.md by replacing it with the sonar quality gate badge.

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/)
 
 ## [unreleased] (Added 🚀 | Changed | Removed 🗑 | Fixed 🐞 | Chore 👨‍💻 👩‍💻)
 
+### Added 🚀
+
+-   Expanded and restructured documentation regarding Docker usage [#3312](https://github.com/MaibornWolff/codecharta/pull/3312) </br>
```
Comment (ce-bo) [inline CHANGELOG.md:12]: Please write changelog entries in present tense according to our changelog guidlines
https://github.com/MaibornWolff/codecharta/blob/main/CONTRIBUTING.md

File: analysis/README.md [hunk 5]
```diff
@@ -61,7 +70,21 @@ This installs all binaries to run the analysis. Java 11 is recommended, while Ja
...
-## Installation
+## Installation via Docker
+
+You can use Codecharta analysis in multiple ways. This section will deal with how to use the analysis as a standalone container. For information on how to use with docker compose, please check out our [docker compose documentation](https://maibornwolff.github.io/codecharta/docs/docker-containers/).
+We assume that you already installed docker, if not, you have to do that before!
+
+To containerize the analysis, please follow the below listed steps.
+
+-   Navigate into the directory you want to analyse with CodeCharta. There are multiple ways to use the docker image:
+    1. Start the docker container and a bash shell in it with `docker run --name codecharta-analysis -it -v $(pwd):$(pwd) -w $(pwd) codecharta/codecharta-analysis bash`. This runs the image, names it `codecharta-analysis` and mounts the current directory and sets it as the working directory in the image. You can now use the codecharta shell or any other of the tools installed in the container via the command line.
```
Comment (ce-bo) [inline analysis/README.md:81]:
```suggestion
    1. Start the docker container and a bash shell in it with `docker run --name codecharta-analysis -it -v $(pwd):$(pwd) -w $(pwd) codecharta/codecharta-analysis bash`. This runs the image, names the container `codecharta-analysis`, mounts the current directory and sets it as the working directory of your container. You can now use the codecharta shell or any other of the tools installed in the container via the command line.
```

File: analysis/README.md [hunk 6]
```diff
@@ -61,7 +70,21 @@ This installs all binaries to run the analysis. Java 11 is recommended, while Ja
...
+    2. Start the docker container and directly use some command (like the ccsh): `docker run --name codecharta-analysis -it -v $(pwd):$(pwd) -w $(pwd) codecharta/codecharta-analysis ccsh`. This starts the Ccsh without any commands, which will open an interactive codecharta shell that will guide you through the parsing/analyzing process. Your current directory will be your previous working directory.
```
Comment (ce-bo) [inline analysis/README.md:82]: I don't understand this sentece `Your current directory will be your previous working directory.`.

Comment (moritz-suckow) [reply analysis/README.md:82]: I was trying to say that the previous working directory of the terminal used to run the command, will be the working directory of the started container. I will change the sentence so this gets more clear

File: CHANGELOG.md [hunk 7]
```diff
@@ -9,15 +9,17 @@ and this project adheres to [Semantic Versioning](http://semver.org/)
...
+-   Add metric tooltips that display attribute descriptors and provide hyperlinks in the sidebar to the metric's documentation [#3273](https://github.com/MaibornWolff/codecharta/pull/3273) </br>
+    <img src="https://user-images.githubusercontent.com/65733509/241383211-d9e8e54b-6b06-45bb-8b99-81cc8e0a4596.png" width="450px"/> <img src="https://github.com/MaibornWolff/codecharta/assets/65733509/0ade9ad4-e60b-4911-aadc-d8142167b21a" width="300px"/>
+-   Expand and restructure documentation regarding Docker usage [#3312](https://github.com/MaibornWolff/codecharta/pull/3312) </br>
 
 ## [1.117.0] - 2023-05-19
 
 ### Added 🚀
 
--   Added support for parser suggestions to SVN-, MetricGardener-, RawText- and SourceCodeParser [#3287](https://github.com/MaibornWolff/codecharta/pull/3287) </br>
+-   Add support for parser suggestions to SVN-, MetricGardener-, RawText- and SourceCodeParser [#3287](https://github.com/MaibornWolff/codecharta/pull/3287) </br>
```
Comment (ce-bo) [inline CHANGELOG.md:20]: Do we need those <br> tags?

Comment (ce-bo) [reply CHANGELOG.md:20]: I think this is only needed when having an img following

File: CHANGELOG.md [hunk 8]
```diff
@@ -9,15 +9,17 @@ and this project adheres to [Semantic Versioning](http://semver.org/)
 
 ### Added 🚀
 
--   Expanded and restructured documentation regarding Docker usage [#3312](https://github.com/MaibornWolff/codecharta/pull/3312) </br>
+-   Add metric tooltips that display attribute descriptors and provide hyperlinks in the sidebar to the metric's documentation [#3273](https://github.com/MaibornWolff/codecharta/pull/3273) </br>
+    <img src="https://user-images.githubusercontent.com/65733509/241383211-d9e8e54b-6b06-45bb-8b99-81cc8e0a4596.png" width="450px"/> <img src="https://github.com/MaibornWolff/codecharta/assets/65733509/0ade9ad4-e60b-4911-aadc-d8142167b21a" width="300px"/>
+-   Expand and restructure documentation regarding Docker usage [#3312](https://github.com/MaibornWolff/codecharta/pull/3312) </br>
```
Comment (ce-bo) [inline CHANGELOG.md:14]: <br> ?

**Review by ce-bo — APPROVED** [review]
LGTM!

File: CHANGELOG.md (hunk 7 above)
Comment (moritz-suckow) [reply CHANGELOG.md:20]: No we do not need them, they are just there because I forgot deleting them after copy pasting.

File: CHANGELOG.md (hunk 8 above)
Comment (moritz-suckow) [reply CHANGELOG.md:14]: This is already in main, I dont know why the changelog is showing that I added it here?

**Pull request merged**